app/controllers/posts_controller.rb

```diff
@@ -1,12 +1,42 @@
 class PostsController < ApplicationController
+  before_action :set_user
+
   def index
-    @user = User.find(params[:user_id])
     @posts = @user.posts
   end
 
   def show
-    @user = User.find(params[:user_id])
     @post = @user.posts.find(params[:id])
-    @comments = @post.comments
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = Post.new
+  end
+
+  def create
+    @post = @user.posts.build(post_params)
+
+    if @post.save
+      flash[:notice] = 'The post was successfully submitted.'
+      redirect_to user_post_path(@user, @post)
+    else
+      flash[:alert] = "Can not save the post. Errors: #{formatted_errors(@post)}"
+      render :new
+    end
+  end
```

Comment:
- [ ] Please make sure to create a new post on behalf of the `current_user` (method available in the application controller) as requested in the project requirements. Instead of `User` use the method `current_user`.

![image](https://github.com/AlaaAlsalem/blogaapp/assets/83803414/4cbbd7d5-b79a-4c5c-9573-ee0a9d1b9cd8)